apps/rush-lib/src/logic/base/BaseInstallManager.ts

```diff
@@ -263,12 +264,39 @@ export abstract class BaseInstallManager {
     shrinkwrapFile: BaseShrinkwrapFile | undefined
   ): Promise<{ shrinkwrapIsUpToDate: boolean; shrinkwrapWarnings: string[] }>;
 
-  protected abstract canSkipInstall(lastInstallDate: Date): boolean;
-
   protected abstract installAsync(cleanInstall: boolean): Promise<void>;
 
   protected abstract postInstallAsync(): Promise<void>;
 
+  protected canSkipInstall(lastModifiedDate: Date): boolean {
+    // Based on timestamps, can we skip this install entirely?
+    const potentiallyChangedFiles: string[] = [];
+
+    // Consider the timestamp on the node_modules folder; if someone tampered with it
+    // or deleted it entirely, then we can't skip this install
+    potentiallyChangedFiles.push(
+      path.join(this.rushConfiguration.commonTempFolder, RushConstants.nodeModulesFolderName)
+    );
+
+    // Additionally, if they pulled an updated npm-shrinkwrap.json file from Git,
+    // then we can't skip this install
+    potentiallyChangedFiles.push(this.rushConfiguration.getCommittedShrinkwrapFilename(this.options.variant));
+
+    // Add common-versions.json file to the potentially changed files list.
+    potentiallyChangedFiles.push(this.rushConfiguration.getCommonVersionsFilePath(this.options.variant));
+
+    if (this.rushConfiguration.packageManager === 'pnpm') {
```

Comment:
Might also be worth doing a quick comparison of the lockfile in common/temp against the committed one, and if they are different, needs an install. Only works if --frozen-lockfile is used, though.

Reply:
We write that prior to every install (meaning the last modify datetime will always be recent), and it's also intended to be hidden from the dev. Considering these two, I _think_ we should be good leaving this as-is.